sequencer/src/message_compat_tests.rs

```diff
@@ -0,0 +1,243 @@
+            view_number: ViewNumber::genesis(),
+            signatures: Default::default(),
+            _pd: Default::default(),
+        }),
+        GeneralConsensusMessage::TimeoutVote(TimeoutVote {
+            signature: (sender, signature.clone()),
+            data: timeout_data,
+            view_number: ViewNumber::genesis(),
+        }),
+        GeneralConsensusMessage::UpgradeProposal(Proposal {
+            data: UpgradeProposal {
+                upgrade_proposal: upgrade_data.clone(),
+                view_number: ViewNumber::genesis(),
+            },
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+        GeneralConsensusMessage::UpgradeVote(UpgradeVote {
+            signature: (sender, signature.clone()),
+            data: upgrade_data,
+            view_number: ViewNumber::genesis(),
+        }),
+    ];
+    let da_messages = [
+        DaConsensusMessage::DaProposal(Proposal {
+            data: DaProposal {
+                encoded_transactions: payload.encode(),
+                metadata,
+                view_number: ViewNumber::genesis(),
+            },
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+        DaConsensusMessage::DaVote(DaVote {
+            signature: (sender, signature.clone()),
+            data: da_data.clone(),
+            view_number: ViewNumber::genesis(),
+        }),
+        DaConsensusMessage::DaCertificate(DaCertificate {
+            data: da_data.clone(),
+            vote_commitment: da_data.commit(),
+            view_number: ViewNumber::genesis(),
+            signatures: Default::default(),
+            _pd: Default::default(),
+        }),
+        DaConsensusMessage::VidDisperseMsg(Proposal {
+            data: VidDisperseShare::from_vid_disperse(VidDisperse::from_membership(
+                ViewNumber::genesis(),
+                vid_scheme(1).disperse(payload.encode()).unwrap(),
+                &membership,
+            ))
+            .remove(0),
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+    ];
+    let data_messages = [DataMessage::SubmitTransaction(
+        transaction,
+        ViewNumber::genesis(),
+    )];
+
+    let seq_messages = consensus_messages
+        .into_iter()
+        .map(SequencingMessage::General)
+        .chain(da_messages.into_iter().map(SequencingMessage::Da));
+    let messages = seq_messages
+        .map(MessageKind::Consensus)
+        .chain(data_messages.into_iter().map(MessageKind::Data))
+        .map(|kind| Message { kind, sender })
+        .collect::<Vec<Message<SeqTypes>>>();
+
+    // Ensure the current serialization implementation generates the same JSON as the committed
+    // reference.
+    let actual = serde_json::to_value(&messages).unwrap();
+    let expected: Value = serde_json::from_str(include_str!("../../data/messages.json")).unwrap();
```

Comment:
Is there any benefit to including the json file in the test binary here?

Reply:
Yeah if we load it at runtime then we have to run the binary from the root of the repo. If you e.g. run `cargo test` from the `sequencer` sub-directory you would get a confusing failue

Reply:
Actually I guess if I use `CARGO_MANIFEST_DIR` then the only requirement is you run the tests on the same machine where they were built...which seems reasonable for tests. I can do that to make the binary smaller/faster to build

Reply:
[111fe29](https://github.com/EspressoSystems/espresso-sequencer/pull/1486/commits/111fe2947c02b705bd7675713c3997bfe1499e09)